File: apiwznm/WznmQPstList.cpp
/**
	* \file WznmQPstList.cpp
	* API code for table TblWznmQPstList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQPstList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQPstList
 ******************************************************************************/

WznmQPstList::WznmQPstList(
			const uint jnum
			, const string sref
			, const string Title
			, const string stubRefWznmMVersion
			, const string srefRefIxVTbl
			, const string titRefIxVTbl
			, const string stubRefUref
			, const string srefIxWznmWArgtype
			, const string titIxWznmWArgtype
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->Title = Title;
	this->stubRefWznmMVersion = stubRefWznmMVersion;
	this->srefRefIxVTbl = srefRefIxVTbl;
	this->titRefIxVTbl = titRefIxVTbl;
	this->stubRefUref = stubRefUref;
	this->srefIxWznmWArgtype = srefIxWznmWArgtype;
	this->titIxWznmWArgtype = titIxWznmWArgtype;
};

bool WznmQPstList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQPstList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVersion", "ver", stubRefWznmMVersion);
		extractStringUclc(docctx, basexpath, "srefRefIxVTbl", "ret", srefRefIxVTbl);
		extractStringUclc(docctx, basexpath, "titRefIxVTbl", "ret2", titRefIxVTbl);
		extractStringUclc(docctx, basexpath, "stubRefUref", "reu", stubRefUref);
		extractStringUclc(docctx, basexpath, "srefIxWznmWArgtype", "aty", srefIxWznmWArgtype);
		extractStringUclc(docctx, basexpath, "titIxWznmWArgtype", "aty2", titIxWznmWArgtype);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQPstList
 ******************************************************************************/

ListWznmQPstList::ListWznmQPstList() {
};

ListWznmQPstList::ListWznmQPstList(
			const ListWznmQPstList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQPstList(*(src.nodes[i]));
};

ListWznmQPstList& ListWznmQPstList::operator=(
			const ListWznmQPstList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQPstList(*(src.nodes[i]));

	return *this;
};

ListWznmQPstList::~ListWznmQPstList() {
	clear();
};

void ListWznmQPstList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQPstList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQPstList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQPstList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQPstList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQPstList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQPstList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQPstList.java
/**
	* \file WznmQPstList.h
	* Java API code for record of table TblWznmQPstList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQPstList {
	
	public WznmQPstList(
				int jnum
				, String sref
				, String Title
				, String stubRefWznmMVersion
				, String srefRefIxVTbl
				, String titRefIxVTbl
				, String stubRefUref
				, String srefIxWznmWArgtype
				, String titIxWznmWArgtype
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Title = Title;
		this.stubRefWznmMVersion = stubRefWznmMVersion;
		this.srefRefIxVTbl = srefRefIxVTbl;
		this.titRefIxVTbl = titRefIxVTbl;
		this.stubRefUref = stubRefUref;
		this.srefIxWznmWArgtype = srefIxWznmWArgtype;
		this.titIxWznmWArgtype = titIxWznmWArgtype;
	};

	public int jnum;
	public String sref;
	public String Title;
	public String stubRefWznmMVersion;
	public String srefRefIxVTbl;
	public String titRefIxVTbl;
	public String stubRefUref;
	public String srefIxWznmWArgtype;
	public String titIxWznmWArgtype;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQPstList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			stubRefWznmMVersion = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVersion", "ver", null, 0);
			srefRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "srefRefIxVTbl", "ret", null, 0);
			titRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "titRefIxVTbl", "ret2", null, 0);
			stubRefUref = Xmlio.extractStringUclc(doc, basexpath, "stubRefUref", "reu", null, 0);
			srefIxWznmWArgtype = Xmlio.extractStringUclc(doc, basexpath, "srefIxWznmWArgtype", "aty", null, 0);
			titIxWznmWArgtype = Xmlio.extractStringUclc(doc, basexpath, "titIxWznmWArgtype", "aty2", null, 0);
			
			return true;
		};

		return false;
	};
};

